crates/webidl/src/first_pass.rs

```diff
@@ -124,18 +125,29 @@ fn first_pass_operation<'src>(
             Argument::Variadic(_) => return Ok(()),
         }
     }
-    record
-        .interfaces
-        .get_mut(self_name)
-        .unwrap()
-        .operations
-        .entry(id)
-        .and_modify(|operation_data| operation_data.overloaded = true)
-        .or_insert_with(Default::default)
-        .argument_names_same
-        .entry(names)
-        .and_modify(|same_argument_names| *same_argument_names = true)
-        .or_insert(false);
+    let mut ids = vec![id];
+    for special in specials {
+        ids.push(match special {
+            Special::Getter(_) => OperationId::IndexingGetter,
+            Special::Setter(_) => OperationId::IndexingSetter,
+            Special::Deleter(_) => OperationId::IndexingDeleter,
+            Special::LegacyCaller(_) => continue,
+        });
+    }
```

Comment:
I think moving code that handles specials here will not help reusing, as specials can be only on interfaces (not even on interface mixins). So it should be fine to just pass a vector of operations ids into this function?

Reply:
Good idea, updated!